src/models/thread.ts

```diff
@@ -90,7 +90,39 @@ export class Thread extends ReadReceipt<ThreadEmittedEvents, ThreadEventHandlerM
 
     private reEmitter: TypedReEmitter<ThreadEmittedEvents, ThreadEventHandlerMap>;
 
+    /**
+     * The last event in this thread, if we don't yet have this in the timeline.
+     *
+     * When we run processRootEvent (which I think happens during the setting-up
```

Comment:
given this is a doc-comment, you could use `{@link}` to help IDEs know that it is a link to a method, which can help if (for example) the target gets renamed:

```suggestion
     * When we run {@link #processRootEvent} (which I think happens during the setting-up
```

That said, I see it makes it harder for a human to read, so ymmv here. Likewise some of the other method names in this comment.